sklearn/ensemble/iforest.py

```diff
@@ -177,12 +216,15 @@ def fit(self, X, y=None, sample_weight=None):
             max_samples = int(self.max_samples * X.shape[0])
 
         self.max_samples_ = max_samples
-        max_depth = int(np.ceil(np.log2(max(max_samples, 2))))
+
+        self._validate_estimator()
+
         super(IsolationForest, self)._fit(X, y, max_samples,
-                                          max_depth=max_depth,
+                                          max_depth=self.max_depth_,
```

Comment:
I really dislike this API of having to pass `max_depth` to the `_fit` method of Bagging, only because of something specific to Isolation Forest. I would simply set the appropriate depth to `self.base_estimator_` inside of `_validate_estimator` (i.e. `self.base_estimator_.set_params(max_deth=soemthing)`)